### PR TITLE
Huffman encoding implementation in scala

### PR DESCRIPTION
**Pull request by kenpower**

### CODE REVIEW

**Comment by Butt4cak3** [comment]
Thank you for another contribution! As you may already know, we don't have any other Scala developers here. That's why your PR might need a little longer to get reviewed. We'll get to them eventually, but someone has to decide to review code that they might not immediately understand, which is obviously quite time consuming. I'm sure you understand.

File: contents/huffman_encoding/code/scala/huffman_encoding.scala [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  case class Leaf(char: Char, var weight: Int) extends Node
+
+  case class Branch(left: Node, right: Node, var weight: Int) extends Node
+
+  def createTree(phrase: String) = {
+
+    val tree = PriorityQueue[Node]()(Ordering.by(-_.weight))
+    tree ++= phrase
+      .groupBy(identity)
```
Comment (jiegillet) [inline contents/huffman_encoding/code/scala/huffman_encoding.scala]: This `.groupBy` function is cool :)

File: contents/huffman_encoding/code/scala/huffman_encoding.scala [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  case class Leaf(char: Char, var weight: Int) extends Node
+
+  case class Branch(left: Node, right: Node, var weight: Int) extends Node
+
+  def createTree(phrase: String) = {
+
+    val tree = PriorityQueue[Node]()(Ordering.by(-_.weight))
+    tree ++= phrase
```
Comment (jiegillet) [inline contents/huffman_encoding/code/scala/huffman_encoding.scala]: Here, the `++=` means `tree = tree concatenated to list of nodes` right?

Comment (kenpower) [reply contents/huffman_encoding/code/scala/huffman_encoding.scala]: yes, append a collection to this queue

**Comment by jiegillet** [comment]
Looks great, thank you!

**Pull request merged**